### ISSUE TITLE
[Bug?]: Fastify server throws an error when using custom server.ts and enriched graphql yoga server

### ISSUE THREAD

**Issue by jan-stehlik** [redwoodjs/redwood#11153]
### What's not working?

I have a standard `graphql` function that works fine with default setting. As soon as I generate server.ts file using `yarn rw setup server-file` the build throws an `TypeError: Cannot destructure property 'healthCheckId' of 'undefined' as it is undefined` error

![image](https://github.com/user-attachments/assets/5056f050-d8e0-4527-a7a9-9f68114b3d65)

This is caused by my graphql [wrapper](https://github.com/jan-stehlik/rw-repro/blob/main/api/src/functions/graphql.ts#L24-L36) that runs each request within  executionContext (so that I can dynamically assign correlation ids across codebase). If I remove this enrichment and just export return value of `createGraphQLHandler` this error disappears.

This happens only when I generate `server.ts` file, because that relies on [__rw_graphqlOptions](https://github.com/redwoodjs/redwood/blob/main/packages/api-server/src/plugins/graphql.ts#L55) that is not correctly generated unless I export handler like this

```ts
// works
export const handler = createGraphQLHandler({
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,
  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect()
  },
})
```

results in following `api/dist/functions/graphql.js`

```js
// __rw_graphqlOptions exists
const __rw_graphqlOptions = {
  loggerConfig: {
    logger: import_logger.logger,
    options: {}
  },
  directives,
  sdls,
  services,
  onException: () => {
    import_db.db.$disconnect();
  }
};
const __rw_handler = (0, import_graphql_server.createGraphQLHandler)(__rw_graphqlOptions);
```

```ts
// breaks
const myHandler = createGraphQLHandler({
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,
  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect()
  },
})

export const handler = myHandler
```

results in following `api/dist/functions/graphql.js`

```js
// __rw_graphqlOptions doesn't exist
const graphQLHandler = (0, import_graphql_server.createGraphQLHandler)({
  loggerConfig: {
    logger: import_logger.logger,
    options: {}
  },
  directives,
  sdls,
  services,
  onException: () => {
    import_db.db.$disconnect();
  }
});
const __rw_handler = graphQLHandler;
```

### How do we reproduce the bug?

1. clone https://github.com/jan-stehlik/rw-repro/tree/main
2. yarn install
3. yarn rw dev
4. build fails and throws an error

### What's your environment? (If it applies)

```shell
System:
    OS: Linux 6.5 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    Shell: 5.1.16 - /bin/bash
Binaries:
    Node: 20.11.1
    Yarn: 4.1.1
```

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by Josh-Walker-GM** [comment]
Hey @jan-stehlik 👋 

Just to clarify then the issue is with the our logic to extract `__rw_graphqlOptions`. The following works fine:
```ts
export const handler = () => {}
```
but when you split it out into a separate variable and export it then it fails:
```ts
const handlerVariable = () => {}
export const handler = handlerVariable
```

It should be possible for us to enhance our logic here to better handle this.

Thanks for a clear reproduction repo! Awesome you're using redwood too, I hope you're enjoying it beyond this issue!

**Comment by jan-stehlik** [comment]
Hey @Josh-Walker-GM , thanks for the reply. Yes the issue is with [babel-plugin-redwood-graphql-options-extract](https://github.com/redwoodjs/redwood/blob/main/packages/babel-config/src/plugins/babel-plugin-redwood-graphql-options-extract.ts) plugin that is responsible for generating `__rw_graphqlOptions`.

Not sure what the best fix is, it could be to have this plugin handle wider number of cases or it may be that `redwoodFastifyGraphQLServer` doesn't rely on generated `__rw_graphqlOptions` but perhaps takes options only as a parameter

**Comment by Josh-Walker-GM** [comment]
Okay leave it with me and I'll see if we can make the plugin smarter to handle this better. I hope to get to this issue this week so that would be my timeline on this one right now.

I assume you aren't blocked by this as you can adopt the other syntax at least until we get a fix in this for this?

**Comment by jan-stehlik** [comment]
perfect, thanks a lot @Josh-Walker-GM , really appreciate it :clap: I'm not completely blocked, but would not want to release my changes until the fix is out as we would lose access to correlation ids across our observebality stack (that's number 1 reason why we use this [wrapper](https://github.com/jan-stehlik/rw-repro/blob/main/api/src/functions/graphql.ts#L24-L36)). 

Thanks for picking this up so quickly!

**Comment by jan-stehlik** [comment]
thanks for fixing so quickly @Josh-Walker-GM :rocket: is this going to be part of next patch release?

**Comment by Josh-Walker-GM** [comment]
Hey @jan-stehlik! Sorry for the delay in getting this one out. I had been hoping to get away with putting it in our v8 release which is currently scheduled to go out next week. It should already be in the v8 rc release. 

If this is really holding you pack I'm happy to explore getting this specific change out in an earlier patch release.
